src/mgl-pax.el

```diff
@@ -177,13 +179,14 @@ other mgl-pax commands in interactive use."
 ;;; If the ASDF-SYSTEM is `:mgl-pax/document' and `(mgl-pax-use-w3m)`
 ;;; is nil, then `:mgl-pax/web' is loaded instead and the web server
 ;;; is started.
-(cl-defmacro mgl-pax-with-component ((asdf-system) &body body)
+(cl-defmacro mgl-pax-with-component
+    ((asdf-system &optional (interactivep `(called-interactively-p 'any))) &body body)
   (declare (indent 1))
   `(let* ((asdf-system (if (and (eq ',asdf-system :mgl-pax/document)
                                 (not (mgl-pax-use-w3m)))
                            :mgl-pax/web
                          ',asdf-system))
-          (interactivep (called-interactively-p 'any))
+          (interactivep ,interactivep)
```

Comment:
Why the rebinding?

Reply:
It isn't a rebinding - the binding of `interactivep` on the lambda-list exists only inside the macro's lexical context, that binds the result of evaluating the form within the macro's binding within the code returned by the macro.